src/inverted_index.c

```diff
@@ -771,6 +771,10 @@ SKIPPER(seekRawDocIdsOnly) {
   // than delta. If it is the last value smaller than delta, we need to skip to the next value.
   if (curVal < delta) {
     cur++;
+    if (cur >= br->buf->offset / 4) {
```

Comment:
Any explanation for this?

Reply:
This was the original code that triggered these PRs. We had this code before #5364 with comments `TODO: explain why it is so`, and `TODO: consider adding a fix`.
Before the PRs we are reverting here, `skipToBlock` may not skip the right block.
We need to revert the support for not being on the right block

Reply:
Yeah I saw this was with that nice `TODO: explain..` before, was wondering if we have a good explanation for it now we can enhance this line with some minimal explanation.